src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder_GlobalSuppressions.cs

```diff
@@ -254,42 +238,114 @@ static bool SupportsGlobalSuppression(ISymbol symbol)
 
         private static bool TryGetExpectedDocumentationCommentId(
             string id,
-            [NotNullWhen(true)] out string? docCommentId)
+            out ReadOnlyMemory<char> docCommentId)
         {
-            if (!ValidateAndSplitDocumentationCommentId(id, out _, out var idPartBeforeArguments, out var arguments))
+            return ValidateAndSplitDocumentationCommentId(id, out _, out docCommentId);
+        }
+
+        /// <summary>
+        /// Validate and split a documentation comment ID into a prefix and complete symbol ID. For the
+        /// <paramref name="docCommentId"/> <c>~M:C.X(System.String)</c>, the <paramref name="prefix"/> would be
+        /// <c>~M:</c> and <paramref name="id"/> would be <c>C.X(System.String)</c>.
+        /// </summary>
+        [PerformanceSensitive("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1224834", Constraint = "Avoid Regex splitting due to high allocation costs.")]
+        private static bool ValidateAndSplitDocumentationCommentId(
+            [NotNullWhen(true)] string? docCommentId,
+            out ReadOnlyMemory<char> prefix,
+            out ReadOnlyMemory<char> id)
```

Comment:
curious why ROM instead of ROS?

Reply:
`ReadOnlySpan<char>` cannot be assigned to a local variable in an `async` method (which the caller is).